### PR TITLE
Improve raider spawn points to be more variable

### PR DESCRIPTION
**Pull request by someaddons**
# Changes proposed in this pull request:
Improve raider spawn points to be more variable
Fix pirates only rarely spawning because liquid wasnt considered a good spawn pos

Review please

### CODE REVIEW

File: src/main/java/com/minecolonies/coremod/colony/buildings/modules/TavernBuildingModule.java [hunk 1]
```diff
@@ -173,7 +175,11 @@ private void spawnVisitor()
 
         newCitizen.getCitizenSkillHandler().init(recruitLevel);
 
-        BlockPos spawnPos = BlockPosUtil.findAround(building.getColony().getWorld(), building.getPosition(), 1, 1, bs -> bs.getMaterial() == Material.AIR);
```
Comment (Raycoms) [inline src/main/java/com/minecolonies/coremod/colony/buildings/modules/TavernBuildingModule.java:176]: I wonder if we want a util for this "findCitizenSpawnPos" and use that then everywhere, also for recall etc

File: src/main/java/com/minecolonies/coremod/colony/managers/RaidManager.java [hunk 2]
```diff
@@ -461,23 +489,52 @@ private BlockPos findSpawnPointInDirections(final BlockPos center, final Directi
...
                     }
                     else
                     {
                         break;
                     }
                 }
-            }
 
+                // Direction two
+                if (random.nextBoolean())
```
Comment (Raycoms) [inline src/main/java/com/minecolonies/coremod/colony/managers/RaidManager.java:507]: I wonder if there is a good way to encapsulate this into a method, seems to be almost 4 times the same code

File: src/main/java/com/minecolonies/coremod/colony/buildings/modules/TavernBuildingModule.java (hunk 1 above)
Comment (someaddons) [reply src/main/java/com/minecolonies/coremod/colony/buildings/modules/TavernBuildingModule.java:176]: ye we could add some predefined to the utils for common cases

File: src/main/java/com/minecolonies/coremod/colony/managers/RaidManager.java (hunk 2 above)
Comment (someaddons) [reply src/main/java/com/minecolonies/coremod/colony/managers/RaidManager.java:507]: wondered too, but difficult as its a conditioned assign to two different vars

**Pull request merged**